overlord/snapstate/snapstate.go

```diff
@@ -1218,13 +1218,16 @@ func InstallMany(st *state.State, names []string, userID int, flags *Flags) ([]s
 		return nil, nil, err
 	}
 
+	var transactionLane int
+	if flags.Transactional {
+		transactionLane = st.NewLane()
+	}
 	tasksets := make([]*state.TaskSet, 0, len(installs))
 	for _, sar := range installs {
 		info := sar.Info
 		var snapst SnapState
-		var flags Flags
 
-		flags, err := ensureInstallPreconditions(st, info, flags, &snapst)
+		validatedFlags, err := ensureInstallPreconditions(st, info, *flags, &snapst)
```

Comment:
we probably want a test like TestInstallPathManyDevMode but for InstallMany to check for the flags code here

Reply:
I've pushed now a commit with a new test.